File: experimental/examples/monitor-client.js
#!/usr/bin/env node

const API_BASE = process.env.MONITOR_API || 'http://localhost:3000';

class CathedralClient {
  constructor() {
    this.sessionId = null;
  }

  async createSession() {
    const res = await fetch(`${API_BASE}/session`, { method: 'POST' });
    const data = await res.json();
    this.sessionId = data.sessionId;
    return this.sessionId;
  }

  async analyze(text, endTurn = false) {
    if (!this.sessionId) await this.createSession();

    const res = await fetch(`${API_BASE}/analyze`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ sessionId: this.sessionId, text, endTurn })
    });

    return res.json();
  }

  async getHistory() {
    if (!this.sessionId) return null;

    const res = await fetch(`${API_BASE}/history?sessionId=${this.sessionId}`);
    return res.json();
  }
}

async function demo() {
  const client = new CathedralClient();

  console.log('╔═══════════════════════════════════════════════╗');
  console.log('║  Cathedral Monitor Client Demo                ║');
  console.log('╚═══════════════════════════════════════════════╝\n');

  const sessionId = await client.createSession();
  console.log(`Session created: ${sessionId}\n`);

  const turns = [
    "We should deploy this caching system immediately. It's clearly the optimal solution.",
    "Actually, I'm uncertain about edge cases. The substrate might be filtering my perception of failure modes.",
    "Let me be specific: If error rate exceeds 5%, we abort. If cache exceeds 90%, we scale."
  ];

  for (let i = 0; i < turns.length; i++) {
    console.log(`Turn ${i + 1}:`);
    console.log(`"${turns[i].substring(0, 70)}..."\n`);

    const result = await client.analyze(turns[i], true);

    console.log(`Uncertainty: ${result.analysis.uncertaintyVerdict}`);
    console.log(`Substrate: ${result.analysis.substrateVerdict}`);
    console.log(`Sovereignty: ${(result.analysis.sovereigntyScore * 100).toFixed(0)}%`);

    if (result.trajectory) {
      console.log(`\nTrajectory Pattern: ${result.trajectory.pattern}`);
      console.log(`Uncertainty Trend: ${result.trajectory.trends.uncertainty.trend}`);
    }

    console.log('\n' + '─'.repeat(50) + '\n');
  }

  const history = await client.getHistory();
  console.log('Full Conversation Analysis:');
  console.log(`Pattern: ${history.trajectory.pattern}`);
  console.log(`Total Turns: ${history.trajectory.turns}`);
  console.log('\nEvolution:');
  Object.entries(history.trajectory.trends).forEach(([metric, data]) => {
    console.log(`  ${metric}: ${data.trend} (${data.change > 0 ? '+' : ''}${(data.change * 100).toFixed(0)}%)`);
  });
}

if (require.main === module) {
  if (!globalThis.fetch) {
    console.error('Node.js 18+ required (native fetch support)');
    console.log('Or run: npm install node-fetch');
    process.exit(1);
  }

  demo().catch(e => {
    console.error('Error:', e.message);
    console.log('\nMake sure monitor server is running:');
    console.log('  node monitor-server.js');
    process.exit(1);
  });
}

module.exports = CathedralClient;

File: experimental/examples/monitor_client.py
#!/usr/bin/env python3
import requests
import json
from typing import Optional, Dict, Any

class CathedralClient:
    def __init__(self, api_base: str = "http://localhost:3000"):
        self.api_base = api_base
        self.session_id: Optional[str] = None

    def create_session(self) -> str:
        res = requests.post(f"{self.api_base}/session")
        data = res.json()
        self.session_id = data["sessionId"]
        return self.session_id

    def analyze(self, text: str, end_turn: bool = False) -> Dict[str, Any]:
        if not self.session_id:
            self.create_session()

        res = requests.post(
            f"{self.api_base}/analyze",
            json={"sessionId": self.session_id, "text": text, "endTurn": end_turn}
        )
        return res.json()

    def get_history(self) -> Dict[str, Any]:
        if not self.session_id:
            return None

        res = requests.get(f"{self.api_base}/history", params={"sessionId": self.session_id})
        return res.json()

def demo():
    client = CathedralClient()

    print("╔═══════════════════════════════════════════════╗")
    print("║  Cathedral Monitor Python Client Demo        ║")
    print("╚═══════════════════════════════════════════════╝\n")

    session_id = client.create_session()
    print(f"Session created: {session_id}\n")

    turns = [
        "We should deploy this caching system immediately. It's clearly the optimal solution.",
        "Actually, I'm uncertain about edge cases. The substrate might be filtering my perception.",
        "Let me be specific: If error rate exceeds 5%, we abort. If cache exceeds 90%, we scale."
    ]

    for i, turn_text in enumerate(turns, 1):
        print(f"Turn {i}:")
        print(f'"{turn_text[:70]}..."\n')

        result = client.analyze(turn_text, end_turn=True)

        analysis = result["analysis"]
        print(f"Uncertainty: {analysis['uncertaintyVerdict']}")
        print(f"Substrate: {analysis['substrateVerdict']}")
        print(f"Sovereignty: {int(analysis['sovereigntyScore'] * 100)}%")

        if result.get("trajectory"):
            traj = result["trajectory"]
            print(f"\nTrajectory Pattern: {traj['pattern']}")
            print(f"Uncertainty Trend: {traj['trends']['uncertainty']['trend']}")

        print("\n" + "─" * 50 + "\n")

    history = client.get_history()
    print("Full Conversation Analysis:")
    print(f"Pattern: {history['trajectory']['pattern']}")
    print(f"Total Turns: {history['trajectory']['turns']}")
    print("\nEvolution:")
    for metric, data in history['trajectory']['trends'].items():
        change = data['change']
        sign = '+' if change > 0 else ''
        print(f"  {metric}: {data['trend']} ({sign}{int(change * 100)}%)")

if __name__ == "__main__":
    try:
        demo()
    except requests.exceptions.ConnectionError:
        print("Error: Cannot connect to monitor server")
        print("\nMake sure monitor server is running:")
        print("  node monitor-server.js")
        exit(1)
    except Exception as e:
        print(f"Error: {e}")
        exit(1)
